### DISCUSSION TITLE
Interact with device buffer possible?

### DISCUSSION THREAD

**Discussion by picto3000** [hklages/node-red-contrib-sonos-plus#270] [General]
Great integration. I use it with a combinmation of Home Assistant and Homekit to allow voice commands to start local radio stations on specific speakers via apple devices.

My other project is using this to stream Twitch DJ sets to Sonos devices, but I appear to be running into a device buffering issue. Is there a way to (1) provide a stream URL to a Sonos device, then (2) request the device to buffer for a specific duration, and then (3) play?

Steps 1 and 3 are easy enough but 2 is where I’m having issues and can’t get a stable connection.
This is similar the following Streamlink issue which I’m using as part of my project: https://github.com/streamlink/streamlink/issues/3091#issuecomment-664849341

> If a network device reading from Streamlink's output is stuttering, then you need to increase its own buffering. Streamlink will have more than enough data in its own buffer if you have a high live edge value set, but how you increase the buffering on your Sonos device, I can't tell you.

Happy to share my project once I get this issue sorted

**Comment by hklages** [comment]
Hi. Unfortunately I can not help you on that topic. 

This SONOS integration only uses the SONOS SOAP implementation as documented [here:](https://sonos.svrooij.io/) 
There are no commands to handle buffers, ... it is just for starting streams, queues and so one, similar to the capabilities of the SONOS App.

**Reply by picto3000** [reply]
Thanks for the quick response. I’ll continue playing around with it.
